lib/upsert/merge_function/postgresql.rb

```diff
@@ -107,34 +107,53 @@ def sql
       end
 
       def use_pg_native?
-        server_version >= 95 && unique_index_on_selector?
+        @use_pg_native ||= server_version >= 95 && unique_index_on_selector?
```

Comment:
to memoize a boolean, do

```
def foo
  return @foo if defined?(@foo)
  @foo = boolean
end
```

otherwise it will recalculate constantly on false or null

Reply:
Of course!  I'm notorious for forgetting that.